### PR TITLE
feat: conditional formatting support all numeric fields

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/4152

### Description:

- [x] there should be a metric fields with numeric types in the field dropdown
- [x] conditional formatting should apply to numeric metric values the same way it does to numeric field values

CleanShot 2023-01-16 at 15 52 45@2x

### CODE REVIEW

File: packages/common/src/utils/conditionalFormatting.ts [hunk 1]
```diff
@@ -42,28 +42,30 @@ export const getConditionalFormattingConfig = (
 };
 
 export const hasMatchingConditionalRules = (
-    value: number | undefined,
+    value: number | string | undefined,
     config: ConditionalFormattingConfig | undefined,
 ) => {
     if (!config) return undefined;
 
+    const parsedValue = typeof value === 'string' ? Number(value) : value;
```
Comment (ZeRego) [inline packages/common/src/utils/conditionalFormatting.ts:50]: Would the lodash function `toNumber` simplify the code and cover more edge cases?

Comment (IrakliJani) [reply packages/common/src/utils/conditionalFormatting.ts:50]: @ZeRego  I added this because metric values were strings 🤷  is that normal?

Comment (IrakliJani) [reply packages/common/src/utils/conditionalFormatting.ts:50]: do you think it might be anything else than string or number?

Comment (ZeRego) [reply packages/common/src/utils/conditionalFormatting.ts:50]: metric values should be numbers 🤔

Comment (ZeRego) [reply packages/common/src/utils/conditionalFormatting.ts:50]: Interesting! metric values are strings 🤔 

Screenshot 2023-01-17 at 15 34 17

**Review by ZeRego — APPROVED** [review]

**Pull request merged**